operations/security_and_authentication/Add_LDAP_User_Federation.md

```diff
@@ -205,59 +265,163 @@ LDAP user federation is not currently configured in Keycloak. For example, if it
-          - default: shasta
-          - type: string
+            - default: shasta
+            - type: string
+      ```
+
+3. Prepare to generate Sealed Secrets.
+   
+   Secrets are stored in customizations.yaml as `SealedSecret` resources 
+   (encrypted secrets), which are deployed by specific charts and decrypted by the
+   Sealed Secrets operator. But first, those Secrets must be seeded generated and
+   encrypted.
+
+   1. Mount the PITDATA so that helm charts are available for the re-install (it might already be mounted).
+
+      ```bash
+      ncn-m001# mkdir -pv /mnt/pitdata
+      ncn-m001# mount -L PITDATA /mnt/pitdata
+      ```
+   
+   2. Load the `zeromq` container image required by Sealed Secret Generators.
+
+      > **NOTE:** A properly configured Docker or Podman environment is required.
+
+      ```bash
+      ncn-m001# /mnt/pitdata/${CSM_RELEASE}/hack/load-container-image.sh dtr.dev.cray.com/zeromq/zeromq:v4.0.5
+      ```
+
+   3. Re-encrypt the existing secrets:
+
+      ```bash
+      ncn-m001# /mnt/pitdata/prep/site-init/utils/secrets-reencrypt.sh customizations.yaml \
+      /mnt/pitdata/prep/site-init/certs/sealed_secrets.key /mnt/pitdata/prep/site-init/certs/sealed_secrets.crt
+      ```
+      
+4. Encrypt the static values in the customizations.yaml file after making changes.
+
+   The following command must be run within the site-init directory.
+
+   ```bash
+   ncn-m001# ./utils/secrets-seed-customizations.sh customizations.yaml
+   ```
+
+   Expected output looks similar to:
+
+      ```bash
+      Creating Sealed Secret keycloak-certs
+      Generating type static_b64...
+      Creating Sealed Secret keycloak-master-admin-auth
+      Generating type static...
+      Generating type static...
+      Generating type randstr...
+      Generating type static...
+      Creating Sealed Secret cray_reds_credentials
+      Generating type static...
+      Generating type static...
+      Creating Sealed Secret cray_meds_credentials
+      Generating type static...
+      Creating Sealed Secret cray_hms_rts_credentials
+      Generating type static...
+      Generating type static...
+      Creating Sealed Secret vcs-user-credentials
+      Generating type randstr...
+      Generating type static...
+      Creating Sealed Secret generated-platform-ca-1
+      Generating type platform_ca...
+      Creating Sealed Secret pals-config
+      Generating type zmq_curve...
+      Generating type zmq_curve...
+      Creating Sealed Secret munge-secret
+      Generating type randstr...
+      Creating Sealed Secret slurmdb-secret
+      Generating type static...
+      Generating type static...
+      Generating type randstr...
+      Generating type randstr...
+      Creating Sealed Secret keycloak-users-localize
+      Generating type static...
+      ```
+
+5. Re-apply the cray-keycloak-users-localize Helm chart with the updated customizations.yaml file.
```

Comment:
Before doing this the admin will have to re-apply the cray-keycloak Helm chart. It's similar to this but they should not delete the chart first.
Also, after the new helm chart is applied they'll need to restart the cray-keycloak statefulset.
This is to pick up the new certs.jks file.

Reply:
Also will need to delay between applying the cray-keycloak Helm chart and restarting Keycloak to give the SealedSecret controller time to update the Secret.